t/09-active_probes.t

```diff
@@ -462,3 +462,61 @@ event: target status 'example.com(127.0.0.1:2114)' from 'false' to 'true'
 checking unhealthy targets: nothing to do
 
 
+
+=== TEST 9: active probes, interval is respected
+--- http_config eval
+qq{
+    $::HttpConfig
+
+    server {
+        listen 2114;
+        location = /status {
+            access_by_lua_block {
+                ngx.sleep(0.3)
+                ngx.exit(200)
+            }
+        }
+    }
+}
+--- config
+    location = /t {
+        content_by_lua_block {
+            local we = require "resty.worker.events"
+            assert(we.configure{ shm = "my_worker_events", interval = 0.1 })
+            local healthcheck = require("resty.healthcheck")
+            local checker = healthcheck.new({
+                test = true,
+                name = "testing",
+                shm_name = "test_shm",
+                type = "http",
+                checks = {
+                    active = {
+                        http_path = "/status",
+                        healthy  = {
+                            interval = 1,
+                            successes = 1,
+                        },
+                        unhealthy  = {
+                            interval = 1,
+                            http_failures = 1,
+                        }
+                    },
+                }
+            })
+            ngx.sleep(1) -- active healthchecks might take up to 1s to start
+            local ok, err = checker:add_target("127.0.0.1", 2114, nil, true)
+            ngx.sleep(1) -- wait for the check interval
+            -- checker callback should not be called more than 5 times
+            if checker.checker_callback_count < 5 then
```

Comment:
Not quite sure I follow the logic around how you came to the number `5`. `interval` is set to `1` second and the code has slept for `2` second so shouldn't this be `2`? Or, perhaps in some edge case where the checker ran immediately before the first sleep `3`?

Reply:
it could be 2 or 3 or 4, but I went with 5 as a "pessimistic" number

if the intervals are 1 second, we slept for 1 second but the checker ran more than 5 times something badly wrong occurred, which is what's happening in `release/1.4.1`